celery/platforms.py

```diff
@@ -781,8 +782,54 @@ def check_privileges(accept_content):
         if not all(hasattr(os, attr)
                    for attr in ['getuid', 'getgid', 'geteuid', 'getegid']):
             raise SecurityError('suspicious platform, contact support')
-
-    if not uid or not gid or not euid or not egid:
+    
+    # Get the group database entry for the current user's group and effective group id using grp.getgrgid() method
+    # We must handle the error in the event of gid or egid not being found.
+    try:
+        gid_entry = grp.getgrgid(gid)
+        egid_entry = grp.getgrgid(egid)
+    except KeyError:
+        print('An entry for the specified gid or egid was not found')
+        if ('pickle' in accept_content or
+                'application/x-python-serialize' in accept_content):
+            if not C_FORCE_ROOT:
+                try:
+                    print(ROOT_DISALLOWED.format(
+                        uid=uid, euid=euid, gid=gid, egid=egid,
+                    ), file=sys.stderr)
+                finally:
+                    sys.stderr.flush()
+                    os._exit(1)
+        warnings.warn(RuntimeWarning(ROOT_DISCOURAGED.format(
+            uid=uid, euid=euid, gid=gid, egid=egid,
+        )))    
+
+    # Get the group and effective group name based on gid
+    gid_grp_name = gid_entry[0]
+    egid_grp_name = egid_entry[0]
+    
+    # Create lists to use in validation step later. 
+    gid_list = [gid_grp_name, egid_grp_name]
+    group_list = ['sudo', 'wheel']    
+
+    # Confirm that uid and euid are not 0 (root)
+    if not uid or not euid:
+        if ('pickle' in accept_content or
+                'application/x-python-serialize' in accept_content):
+            if not C_FORCE_ROOT:
+                try:
+                    print(ROOT_DISALLOWED.format(
+                        uid=uid, euid=euid, gid=gid, egid=egid,
+                    ), file=sys.stderr)
+                finally:
+                    sys.stderr.flush()
+                    os._exit(1)
+        warnings.warn(RuntimeWarning(ROOT_DISCOURAGED.format(
+            uid=uid, euid=euid, gid=gid, egid=egid,
+        )))
+    # Confirm that the gid and egid are not one that can be used to escalate privilege
+    # Be sure to have this "if" statement uses 'x in long-list for x in short-list' (group_list will be the longer list)
+    elif any((True for x in group_list if x in gid_list)):
```

Comment:
```suggestion
    elif any(x in gid_list for x in group_list):
```

Python accepts generators without parenthesis as arguments.
Furthermore, the condition is redundant.

Reply:
Also, the variable names are not as meaningful as they could be.

Reply:
Variables renamed as well.